cli/slsa-verifier/main_regression_test.go

```diff
@@ -96,293 +96,297 @@ func Test_runVerifyGHAArtifactPath(t *testing.T) {
 		minversion string
 		// specifying builders will restrict builders to only the specified ones.
 		builders []string
+		// specify provenance path if not the same as artifacts[0]
+		// useful for testing provenance with multiple artifacts,
+		// without needing to duplicate provenance
+		provenancePath string
 	}{
 		{
-			name:     "valid main branch default",
-			artifact: "binary-linux-amd64-workflow_dispatch",
-			source:   "github.com/slsa-framework/example-package",
+			name:      "valid main branch default",
+			artifacts: []string{"binary-linux-amd64-workflow_dispatch"},
+			source:    "github.com/slsa-framework/example-package",
 		},
 		{
 			name:       "valid main branch default - invalid builderID",
-			artifact:   "binary-linux-amd64-workflow_dispatch",
+			artifacts:  []string{"binary-linux-amd64-workflow_dispatch"},
```

Comment:
is it do-able to add a few examples with all the scenarios you describe in the PR description?

Reply:
Yes, it's just that I got confused in the types and broke the build. Should be able to fix it in the next 30mins, sorry.

Apparently I cannot test locally, missing some setup step

Reply:
what happens when you try locally? You may be hitting rate limiting from rekor / sigstore

Reply:
I think that's the case. In any case, I got it working for passing tests (multiple artifacts, different order, repeated artifacts), now I need to handle the failing case.